cbdr/cortex
# COR-691: Refine RSS Feed

## **Purpose:**

> As the curator on Uberflip, I want cortex to pass me one single feed that contains all the metadata from a post, but also the category that it has been filed in.

- Wiped old blog posts
- Originally suspected missing feed items might be due to limiting the feed to only certain categories. That was not the case
  - Verified that our feed does output all categories on a tenant + the category name(s).
  - [See the example XML output](https://gist.github.com/arelia/84c92c9e95eeebb12ca4ed5693b08afe)
 - Saw that feed breaks if a post is missing a featured image
   - Requested a bug to prevent this here: https://trello.com/c/cdAiPNoz/13-cortex-rss-without-featured-media-breaks
   - Special characters can also break the feed, requested a bug: https://trello.com/c/HdhWa4E5/14-need-to-validate-for-characters-in-cortex-fields-that-might-break-the-rss-feed-e-g-periods-in-slugs
- Next, checked feed for validity. Our feed did not validate: https://www.screencast.com/t/B26ZiMxi
  - Validator is expecting `<author>` and not `<author:encoded>`
  - This PR fixes that and I don't anticipate side effects since there's no special formatting or encoding needed on the author field
  - **Also noticed we are not populating the author field**. [See the author line in the sample XML](https://gist.github.com/arelia/84c92c9e95eeebb12ca4ed5693b08afe#file-cortext_prod_2017-04-06-xml-L12)
- Fixing `<author>` allows validation to continue and reveal that **`<pubDate>` value is also not valid**.
   - Need to reformat date for [RFC-822](https://www.feedvalidator.org/docs/error/InvalidRFC2822Date.html) 
   -  Notice in the [sample XML](https://gist.github.com/arelia/84c92c9e95eeebb12ca4ed5693b08afe#file-cortext_prod_2017-04-06-xml-L26) we output `<pubDate>06/04/2017 16:06</pubDate>`
   - Instead should be in this format: `<pubDate>Wed, 02 Oct 2002 08:00:00 EST</pubDate>`

 - Making those changes produces an XML that validates and Uberflip imports it successfully: https://gist.github.com/arelia/2351cbe482599a4b7a9a713a5547f376
   - Validator result: https://screencast.com/t/UDQa7msf
   - Uberflip: https://screencast.com/t/iyjJZik4

## On Prod
**Need to reseed, add new sample posts, import into Uberflip before story can be verified**

## **Follow up Stories:**
 - [Resolve the missing author name and email](https://trello.com/c/Fxht2MWm/15-cortex-rss-resolve-the-missing-author-name-and-email)
 - [Reformat the publish date for RSS](https://trello.com/c/eOi5a497/16-cortex-rss-reformat-publish-date)

## **JIRA:**
https://cb-content-enablement.atlassian.net/browse/COR-691

## **Changes:**
* Changes to setup, Architectural changes, Migrations, Library changes, Side effects, Screenshots
n/a


## **QA Links:**
none, need to complete the follow up stories before we can produce a valid RSS feed

## **How to Verify These Changes**
* Specific pages to visit
n/a

* Steps to take
  * See the XML from the ["Valid Cortex" gist](https://gist.githubusercontent.com/arelia/2351cbe482599a4b7a9a713a5547f376/raw/e2d67eac6bae5227a84c059fa8b80b2d334709da/validCortex.xml) and see that it [passes validation here](https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Fgist.githubusercontent.com%2Farelia%2F2351cbe482599a4b7a9a713a5547f376%2Fraw%2Fe2d67eac6bae5227a84c059fa8b80b2d334709da%2FvalidCortex.xml)
  * This XML URL won't work in Uberflip. You can test by serving this xml locally in the `public` directory and passing an ngrok URL to Uberflip, such as http://c0e65c2c.ngrok.io/blog.xml. You should see your items successfully imported into Uberflip

![image](https://cloud.githubusercontent.com/assets/2359538/24841691/74a2622c-1d4f-11e7-852f-10e0ae6dcfce.png)


## Relevant PRs/Dependencies
Original RSS PR: https://github.com/cbdr/cortex/pull/379

##  **Responsive considerations, Additional Information**
n/a
